### PR TITLE
BitchX: new recipe

### PR DESCRIPTION
**Pull request by extrowerk**

### CODE REVIEW

File: net-irc/bitchx/bitchx-1.2.1.recipe [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	1995, 1998 Jeremy Nelson and others (\"EPIC Software Labs\")
+	1996-2004 Colten Edwards et al.
+	2008-2014 Kevin Easton et al."
+LICENSE="BSD (3-clause)"
+REVISION="1"
+SOURCE_URI="http://www.bitchx.com/download/bitchx-$portVersion.tar.gz"
+CHECKSUM_SHA256="2d270500dd42b5e2b191980d584f6587ca8a0dbda26b35ce7fadb519f53c83e2"
+PATCHES="bitchx-$portVersion.patchset"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
```
Comment (fbrosson) [inline net-irc/bitchx/bitchx-1.2.1.recipe]: Suggestion:
```diff
-ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+ARCHITECTURES="x86_gcc2 ?x86 x86_64"
```
Good news! A tiny patch (swapping lines 999 and 1000 of `bitchx-1.2.1/source/istatus.c`) is enough to make it build with gcc2:
```diff
diff --git a/source/status.c b/source/status.c
index 36bba31..20bccf7 100644
--- a/source/status.c
+++ b/source/status.c
@@ -996,8 +996,8 @@ static      char    *status_notify_windows(Window *window)
 
        if (get_int_var(SHOW_STATUS_ALL_VAR) || window == window->screen->current_window)
        {
-               *notes = 0;
                Window *notify_win = NULL;
+               *notes = 0;
                while ((traverse_all_windows(&notify_win)))
                {
                        if (notify_win->miscflags & WINDOW_NOTIFIED)
```

File: net-irc/bitchx/bitchx-1.2.1.recipe [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+* Ease of use. BitchX adds dozens of useful command aliases to reduce typing.
+* Built-in notify, protection, bot, and shit lists.
+* Built-in mass commands and tools.
+* Extended set of DCC commands and built-in CDCC offering.
+* Built-in link looker.
+* Extended scripting functionality, including functions found nowhere else.
+* Huge installed user base, an abundance of scripts, and a dedicated help \
+channel (#BitchX on EFNET).
+* Built-in screen client, allowing you to detach and re-attach to an irc session."
+HOMEPAGE="http://www.bitchx.com/"
```
Comment (keaston) [inline net-irc/bitchx/bitchx-1.2.1.recipe]: www.bitchx.org is the actual homepage, bitchx.com isn't associated with the project.

File: net-irc/bitchx/patches/bitchx-1.2.1.patchset [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+index 29400c2..b6c2dff 100644
+--- a/include/irc_std.h
++++ b/include/irc_std.h
+@@ -205,9 +205,11 @@ int	killpg (int pgrp, int sig);
+ char *	getpass (const char * prompt);
+ #endif
+ 
++#ifndef __HAIKU__
+ #ifndef BCOPY_DECLARED
+ void	bcopy (const void *src, void *dest, size_t n);
```
Comment (keaston) [inline net-irc/bitchx/patches/bitchx-1.2.1.patchset]: Which header declares bcopy() on Haiku? The configure script looks for it in <string.h>, ideally we'd fix this in configure.

File: net-irc/bitchx/bitchx-1.2.1.recipe (hunk 2 above)
Comment (extrowerk) [reply net-irc/bitchx/bitchx-1.2.1.recipe]: Thanks,haven't noticed it.

File: net-irc/bitchx/patches/bitchx-1.2.1.patchset (hunk 3 above)
Comment (alaviss) [reply net-irc/bitchx/patches/bitchx-1.2.1.patchset]: [`bcopy()`](http://xref.plausible.coop/source/xref/haiku/headers/posix/strings.h#24) in Haiku is a macro, not a function.

File: net-irc/bitchx/bitchx-1.2.1.recipe [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	"
+BUILD_PREREQUIRES="
+	cmd:autoreconf
+	cmd:gcc
+	cmd:grep
+	cmd:make
+	cmd:pkg_config
+	"
+
+defineDebugInfoPackage bitchx\
```
Comment (alaviss) [inline net-irc/bitchx/bitchx-1.2.1.recipe]:
```suggestion
defineDebugInfoPackage bitchx \
```

File: net-irc/bitchx/bitchx-1.2.1.recipe [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	1991, 1992 Troy Rollo
+	1992-1996 Matthew Green
+	1993, 1997 Jeremy Nelson
+	1994 Jake Khuon
+	1995, 1998 Jeremy Nelson and others (\"EPIC Software Labs\")
+	1996-2004 Colten Edwards et al.
+	2008-2014 Kevin Easton et al."
+LICENSE="BSD (3-clause)"
+REVISION="1"
+SOURCE_URI="http://www.bitchx.com/download/bitchx-$portVersion.tar.gz"
```
Comment (alaviss) [inline net-irc/bitchx/bitchx-1.2.1.recipe]:
```suggestion
SOURCE_URI="https://download.sourceforge.net/bitchx/bitchx-$portVersion.tar.gz"
```

File: net-irc/bitchx/patches/bitchx-1.2.1.patchset (hunk 3 above)
Comment (keaston) [reply net-irc/bitchx/patches/bitchx-1.2.1.patchset]: Ahh I see.  In that case you could change

```
-#ifndef BCOPY_DECLARED
+#if !defined(BCOPY_DECLARED) && !defined(bcopy)

```
The best fix upstream is to replace all remaining uses of `bcopy()` with `memmove()`.

File: net-irc/bitchx/bitchx-1.2.1.recipe [hunk 6]
```diff
@@ -12,7 +12,7 @@ Features:
 * Huge installed user base, an abundance of scripts, and a dedicated help \
 channel (#BitchX on EFNET).
 * Built-in screen client, allowing you to detach and re-attach to an irc session."
-HOMEPAGE="http://www.bitchx.com/"
+HOMEPAGE="http://www.bitchx.org"
```
Comment (fbrosson) [inline net-irc/bitchx/bitchx-1.2.1.recipe]: Suggestion: add a trailing slash to [`HOMEPAGE`](https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#homepage)
```diff
-HOMEPAGE="http://www.bitchx.org"
+HOMEPAGE="http://www.bitchx.org/"
```

**Comment by alaviss** [comment]
I've just realized that I haven't check the user configuration location. Can you help us @extrowerk?

**Pull request merged**